File: frontend/src/types/supplier.ts
export interface Supplier {
    id: number;
    code: string;
    name: string;
    name_en?: string;
    tax_code?: string;
    contact_person?: string;
    phone?: string;
    email?: string;
    address?: string;
    city?: string;
    country: string;
    payment_terms?: string;
    credit_limit?: number;
    is_active: boolean;
    notes?: string;
    created_at: string;
    updated_at: string;
}

export interface CreateSupplierInput {
    code: string;
    name: string;
    name_en?: string;
    tax_code?: string;
    contact_person?: string;
    phone?: string;
    email?: string;
    address?: string;
    city?: string;
    country?: string;
    payment_terms?: string;
    credit_limit?: number;
    is_active?: boolean;
    notes?: string;
}

export interface UpdateSupplierInput {
    code?: string;
    name?: string;
    name_en?: string;
    tax_code?: string;
    contact_person?: string;
    phone?: string;
    email?: string;
    address?: string;
    city?: string;
    country?: string;
    payment_terms?: string;
    credit_limit?: number;
    is_active?: boolean;
    notes?: string;
}

export interface SupplierFilters {
    search?: string;
    country?: string;
    city?: string;
    is_active?: boolean;
    page?: number;
    page_size?: number;
    sort_by?: string;
    sort_order?: 'asc' | 'desc';
}

export interface SupplierListResponse {
    success: boolean;
    data: Supplier[];
    pagination: {
        page: number;
        page_size: number;
        total: number;
        total_pages: number;
    };
}

File: backend/internal/models/supplier.go
package models

import (
	"time"
)

// Supplier represents a supplier entity
type Supplier struct {
	ID            uint       `json:"id" gorm:"primaryKey"`
	Code          string     `json:"code" gorm:"uniqueIndex;not null"`
	Name          string     `json:"name" gorm:"not null"`
	NameEn        *string    `json:"name_en"`
	TaxCode       *string    `json:"tax_code"`
	ContactPerson *string    `json:"contact_person"`
	Phone         *string    `json:"phone"`
	Email         *string    `json:"email"`
	Address       *string    `json:"address"`
	City          *string    `json:"city"`
	Country       string     `json:"country" gorm:"default:'Vietnam'"`
	PaymentTerms  *string    `json:"payment_terms"`
	CreditLimit   *float64   `json:"credit_limit" gorm:"type:decimal(15,2)"`
	IsActive      *bool      `json:"is_active" gorm:"default:true"`
	Notes         *string    `json:"notes"`
	CreatedAt     time.Time  `json:"created_at"`
	CreatedBy     *uint      `json:"created_by"`
	UpdatedAt     time.Time  `json:"updated_at"`
	UpdatedBy     *uint      `json:"updated_by"`
}

// TableName specifies the table name for Supplier
func (Supplier) TableName() string {
	return "suppliers"
}

// SafeSupplier represents supplier data safe for API responses
type SafeSupplier struct {
	ID            uint      `json:"id"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	NameEn        *string   `json:"name_en"`
	TaxCode       *string   `json:"tax_code"`
	ContactPerson *string   `json:"contact_person"`
	Phone         *string   `json:"phone"`
	Email         *string   `json:"email"`
	Address       *string   `json:"address"`
	City          *string   `json:"city"`
	Country       string    `json:"country"`
	PaymentTerms  *string   `json:"payment_terms"`
	CreditLimit   *float64  `json:"credit_limit"`
	IsActive      *bool     `json:"is_active"`
	Notes         *string   `json:"notes"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// ToSafe converts Supplier to SafeSupplier
func (s *Supplier) ToSafe() *SafeSupplier {
	return &SafeSupplier{
		ID:            s.ID,
		Code:          s.Code,
		Name:          s.Name,
		NameEn:        s.NameEn,
		TaxCode:       s.TaxCode,
		ContactPerson: s.ContactPerson,
		Phone:         s.Phone,
		Email:         s.Email,
		Address:       s.Address,
		City:          s.City,
		Country:       s.Country,
		PaymentTerms:  s.PaymentTerms,
		CreditLimit:   s.CreditLimit,
		IsActive:      s.IsActive,
		Notes:         s.Notes,
		CreatedAt:     s.CreatedAt,
		UpdatedAt:     s.UpdatedAt,
	}
}
